doc/radosgw/keycloak.rst

```diff
@@ -125,3 +125,104 @@ The tags can be verified using token introspection of the JWT. The command to in
     -d "token=$KC_ACCESS_TOKEN" \
     "http://$KC_SERVER/$KC_CONTEXT/realms/$KC_REALM/protocol/openid-connect/token/introspect" \
     | jq .
+
+Example showing how to fetch a web token from Keycloak using Kerberos or Username and Password authentication
+=============================================================================================================
+
+The following python script takes various command line arguments to authenticate against Keycloak via Kerberos or Basic authentication to obtain a Java Web 
+Token. This token can then be exchanged for temporary credentials with the RADOS gateway in order to AssumeRoleWithWebIdentity:
+
+.. code-block:: python
+
+    #!/usr/bin/env python
+    import json,requests,urllib,re,argparse,sys,os,boto3
+    from requests_kerberos import HTTPKerberosAuth, OPTIONAL
+    from lxml import html
+    from getpass import getpass
+    
+    def get_token(client,client_secret,redirect,base_url,auth_uri,token_uri,method,verify_ssl,user,password):
+        session = requests.session()
+        if method == 'kerberos':
+          kerberos_auth = HTTPKerberosAuth(mutual_authentication=OPTIONAL, force_preemptive=True)
+          access_code_params = {'client_id': client, 'response_type': 'code', 'scope': 'openid', 'redirect_uri': redirect}
+          try:
+            get_auth_response = session.get(base_url + auth_uri + "?" + urllib.parse.urlencode(access_code_params), verify=verify_ssl, auth=kerberos_auth, allow_redirects=False)
+          except Exception as e:
```

Comment:
Probably better to avoid the `try` and bare `except Exception` here, and just let the exceptions bubble up.